### PR TITLE
Add functions for creating sm domain

### PR DESCRIPTION
**Pull request by alprinf**
PR for adding functions needed for Sagemaker domain creation
- added `get_subnet_ids_with_vpc_id` function and test case

### CODE REVIEW

File: baram/ec2_manager.py [hunk 1]
```diff
@@ -299,6 +299,9 @@ def list_detail_subnets(self) -> list:
                  'state': subnet['State']})
         return subnet_list
 
+    def get_subnet_ids_with_vpc_id(self, vpc_id: str):
```
Comment (lks21c) [inline baram/ec2_manager.py]: please add return type hint and description comment.

File: tests/test_ec2_manager.py [hunk 2]
```diff
@@ -256,6 +256,17 @@ def test_list_detail_subnets(em):
...
+def test_get_subnet_ids_with_vpc_id(em):
+    # Given
+    default_vpc_id = em.get_default_vpc()['VpcId']
+
+    # When
+    subnets = em.get_subnet_ids_with_vpc_id(default_vpc_id)
+
+    # Then
+    assert len(subnets) > 0
+    pprint(subnets)
```
Comment (lks21c) [inline tests/test_ec2_manager.py]: it would be better to use `print(json.dumps(subents, indent=4))` since it's able to inject online json parser directly.

**Review by lks21c — CHANGES_REQUESTED** [review]

File: baram/ec2_manager.py (hunk 1 above)
Comment (alprinf) [reply baram/ec2_manager.py]: sure I will add them

File: tests/test_ec2_manager.py (hunk 2 above)
Comment (alprinf) [reply tests/test_ec2_manager.py]: thanks for suggestion. will use json dumps!

**Review by lks21c — APPROVED** [review]

**Comment by lks21c** [comment]
merge after 1 approval to speed up development.

**Pull request merged**